janMarkusMilan/assignments/slidingwindow/lc340/lc340.js

```diff
@@ -0,0 +1,110 @@
+// Problem Statement #
+
+// Given a string, find the length of the longest substring in it with no more than K distinct characters.
+
+// You can assume that K is less than or equal to the length of the given string.
+
+/*
+Hashmap: Keep track of every instance of each character
+Var: max
+Loop (forms a window):
+  add the char to the hashmap (if it is in hashmap, increment)
+  if occurrence of a char > K:
+    slide window
+  update length to max
+
+return the max
+
+'araaci'
+{}
+Iteration 1: {'a': 1}
+Iteration 2: {'a': 1, 'r': 1}
+Iteration 3: {'a': 2, 'r': 1}
+Iteration 4: {'a': 3, 'r': 1}
+Iteration 5: {'a': 3, 'r': 1, 'c': 1}
+Iteration 6: {'a': 3, 'r': 1, 'c': 1, 'i': 1}
```

Comment:
Love that you are demonstrating your approach using a test case! However, for the if statement in your pseudo-code, it is unclear how "occurrence of a char" is calculated or determined. Writing out the code using sentences and being detailed will help the interviewer understand your approach.